example/service.twirp.go

```diff
@@ -45,24 +45,31 @@ type Haberdasher interface {
 type haberdasherProtobufClient struct {
 	client HTTPClient
 	urls   [1]string
+	opts   twirp.ClientOptions
 }
 
 // NewHaberdasherProtobufClient creates a Protobuf client that implements the Haberdasher interface.
 // It communicates using Protobuf and can be configured with a custom HTTPClient.
-func NewHaberdasherProtobufClient(addr string, client HTTPClient) Haberdasher {
+func NewHaberdasherProtobufClient(addr string, client HTTPClient, opt ...twirp.ClientOption) Haberdasher {
+	var httpClient HTTPClient = client
+	if c, ok := client.(*http.Client); ok {
+		httpClient = withoutRedirects(c)
+	}
```

Comment:
The `httpClient` variable is unnecessary. You can do `client = withoutRedirects(c)` inside the braces.